config/main.py

```diff
@@ -167,11 +167,42 @@ def validate_namespace(namespace):
     else:
         return False
 
-def interface_alias_to_name(interface_alias):
+# Return the namespace where an interface belongs
+def get_intf_namespace(port):
+    """If it is a non multi-asic device, or if the interface is management interface
+       return '' ( DEFAULT_NAMESPACE ) which maps to current namespace ( in case of config commands
+       it is linux host )
+    """
+    if sonic_device_util.is_multi_npu() == False or port == 'eth0':
+        return DEFAULT_NAMESPACE
+
+    # If it is PortChannel or Vlan interface or Loopback, user needs to input the namespace.
+    if port.startswith("PortChannel") or port.startswith("Vlan") or port.startswith("Loopback"):
```

Comment:
no need to hardcode all these logics.

Reply:
The idea of checking for interfaces like PortChannel, SVI interfaces here and return back "None" explicitly so that the user will have to mandatorily pass the parameter -n namespace for such interfaces.

For physical interfaces ( those starting with "Ethernet") idea was to make the -namespace parameter optional. 
  * If the user don't provide it, the config handler will use the namespace obtained from this get_port_namespace() API.
  * If the user provides the namespace parameter,  the config handler will make sure it is same as what we find from this get_port_namespace() API.

For the physical interfaces behavior could be similar to single NPU devices now .. where user need not pass the namespace as a mandatory parameter while configuring.

Reply:
Added a new API get_port_table_name() to get the table to be checked for the existence of the interface.